src/withStoreSubscription.tsx

```diff
@@ -0,0 +1,37 @@
+/**
+* withStoreSubscription.ts
+* Author: Brent Traut
+* Copyright: Microsoft 2017
+*
+* withStoreSubscription is a HOC that tries to reduce the store subscription boilerplate
+* code. It moves autosubscription values out of state and into props by inserting an extra
+* container component.
+*
+* Create a wrapper factory by passing a buildState function into withStoreSubscription.
+* Then pass dumb components into the wrapper factory to wrap them with autosubscribed
+* container components that leverage the given buildState method.
+*/
+
+'use strict';
+
+import * as React from 'react';
+import ComponentBase from './ComponentBase';
+
+export function withStoreSubscription<
+    WrapperProps extends React.Props<any>,
```

Comment:
How does this work with refs? One issue with HOC is that ref isn’t a prop that can be passed through to the wrapped component (as I understand it). The ref will point to the wrapper and not the component which can be problematic for direct manipulation (calling a public method on the component) or, potentially, when measuring the component rect in layout (this may not actually be a problem, I’m speculating).

Reply:
I suppose that's a potential concern, but calling direct manipulation on child components sounds very un-Reacty. Generally all communication down the tree should come through props (and context) and all upward should come through callbacks. Do you have an example where refs are necessary?

Reply:
Note also that many extremely popular React libraries use the HOC pattern which happens to be part of my motivation for introducing this pattern to ReSub. It's pretty well established and battle-tested.

Reply:
One example that comes to mind is calling focus() on a specific component for accessibility reasons. It may not be the norm, but it would be nice to be able to pass a ref callback that points at the wrapped component vs the wrapper.

Reply:
I don't think that's a good pattern anyway. You still need to operate on native nodes for accessibility purposes. That way calling ```ReactDOM.findDOMNode()``` will unwrap the HOC just as well.